jujugui/static/gui/src/app/models/models.js

```diff
@@ -312,6 +312,64 @@ YUI.add('juju-models', function(Y) {
     },
 
     /**
+    Given a relation between two services (this one and one other), return the
+    service on the other end of the relation.
+
+    @method getOtherServiceFromRelation
+    @param relation The relation to check
+    @param db The database of services and relations
+    @return The other service in the relation
+    */
+    getOtherServiceFromRelation: function(relation, db) {
+      var endpoints = relation.get('endpoints');
+      if (!endpoints || endpoints.length !== 2) {
+        return;
+      }
+      if (endpoints[0][0] === this.get('id')) {
```

Comment:
It's not really clear what's going on here...I'm guessing you're saying if this endpoint is the same as the id then use another? Could you add a code comment to explain this for future us?